src/main/java/filemanager/launcher/Launcher.java

```diff
@@ -3,13 +3,22 @@
 import com.google.inject.Guice;
 import com.google.inject.Injector;
 import filemanager.binder.FileServiceBinderModule;
-import filemanager.controller.Application;
+import filemanager.configuration.FileHandlerConfiguration;
+import filemanager.directorytracker.ScheduleFileDirectory;
+import filemanager.managed.ScheduleFileDirectoryManaged;
+import io.dropwizard.Application;
+import io.dropwizard.setup.Environment;
 
-public class Launcher {
-    public static void main(String[] args) {
-        Injector guice = Guice.createInjector(new FileServiceBinderModule());
-        Application application = guice.getInstance(Application.class);
-        application.runProgram();
+public class Launcher extends Application<FileHandlerConfiguration> {
+    public static void main(String[] args) throws Exception {
+        new Launcher().run(args);
     }
 
+    @Override
+    public void run(FileHandlerConfiguration configuration, Environment environment) {
+        Injector guice = Guice.createInjector(new FileServiceBinderModule());
+        ScheduleFileDirectory trackerFileDirectory = guice.getInstance(ScheduleFileDirectory.class);
+        ScheduleFileDirectoryManaged managed = new ScheduleFileDirectoryManaged(trackerFileDirectory, configuration);
+        environment.lifecycle().manage(managed);
```

Comment:
There is a better way to have managed scheduled executor service in Dropwizard ;) lifecycle().scheduledExecutorService()
Please try using this one instead of writing wrapper class to accomplish that function.